pkg/tests/cross-tests/input_cross_test.go

```diff
@@ -219,3 +219,27 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
 		}),
 	})
 }
+
+func TestOptionalSetNotSpecified(t *testing.T) {
+	skipUnlessLinux(t)
+
+	t.Skip("We misrepresent empty sets")
+	// TODO[pulumi/pulumi-terraform-bridge#1970]
+
+	runCreateInputCheck(t, inputTestCase{
+		Resource: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"f0": {
+					Optional: true,
+					Type:     schema.TypeSet,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"x": {Optional: true, Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+		Config: tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{}),
```

Comment:
This is very strange, it's not that the set is empty, it's that it's not defined at all. So it should be treated as missing (which per Optional: true should be fine). Any more logs as to what's going on differently here? Certain these are not test-level issues ?

Reply:
BTW. I'm not sure this `tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{}),` is right at all. We need tftypes.Object to match 

```
Schema: map[string]*schema.Schema{
				"f0": {
					Optional: true,
					Type:     schema.TypeSet,
					Elem: &schema.Resource{
						Schema: map[string]*schema.Schema{
							"x": {Optional: true, Type: schema.TypeString},
						},
					},
				},
			},
```

Even if it is empty. That is the normal situation.

I think the framework will let you write the value as `map[string]any{}` and it will infer the correct tftypes.Object if you do so.

Reply:
I think it's what https://github.com/pulumi/pulumi-terraform-bridge/issues/1964 is hitting.

pulumi program:

```yaml
backend:
    url: file://./data
name: project
resources:
    example:
        properties: {}
        type: crossprovider:index:TestRes
runtime: yaml
```

TF program

```terraform
resource "crossprovider_testres" "example" {
}
```

Reply:
The program seem fine to me, so it looks like a bridge issue

Reply:
Fair, the program looks fine. What is the bridge issue. Can you attach logs etc.

Reply:
I haven't dug into it yet, will let you know once I have something more.

Reply:
I think it happens here: https://github.com/pulumi/pulumi-terraform-bridge/blob/7aa41da657b39cb1791d936bd53d6fe06dc3d52c/pkg/tfshim/sdk-v2/cty.go#L232